client/packages/system/src/Stock/Components/StockLineForm.tsx

```diff
@@ -296,37 +314,57 @@ export const StockLineForm = ({
               />
             }
           />
+          {showItemVariantsInput && (
+            <StyledInputRow
+              label={t('label.item-variant')}
+              Input={
+                <ItemVariantInput
+                  itemId={draft.itemId}
+                  selectedId={draft?.itemVariant?.id}
+                  width={160}
+                  disabled={!packEditable}
+                  onChange={variant => {
+                    const packaging = variant?.packagingVariants.find(
+                      p => p.packSize === draft.packSize
+                    );
+                    const volumePerPack =
+                      ((packaging?.volumePerUnit ?? 0) / 1000) *
+                      (draft?.packSize ?? 1);
+
+                    onUpdate({
+                      itemVariant: variant,
+                      volumePerPack,
+                    });
+                  }}
+                />
+              }
+            />
+          )}
           <StyledInputRow
-            label={t('label.barcode')}
+            label={t('label.volume-per-pack')}
             Input={
-              <Box style={{ width: 162 }}>
-                <BufferedTextInput
-                  value={draft.barcode ?? ''}
-                  onChange={e => onUpdate({ barcode: e.target.value })}
-                />
-                {isEnabled && (
-                  <Tooltip
-                    title={isConnected ? '' : t('error.scanner-not-connected')}
-                  >
-                    <Box>
-                      <IconButton
-                        disabled={isScanning || !isConnected}
-                        onClick={scanBarcode}
-                        icon={
-                          isScanning ? (
-                            <CircularProgress size={20} color="secondary" />
-                          ) : (
-                            <ScanIcon />
-                          )
-                        }
-                        label={t('button.scan')}
-                      />
-                    </Box>
-                  </Tooltip>
-                )}
-              </Box>
+              <NumericTextInput
+                autoFocus
+                disabled={!packEditable}
```

Comment:
So will only be able to edit volume per pack when adding stock for the first time? If its wrong, how to adjust? 

Especially when repacking to a different pack size?

Reply:
Can make it editable... repack already auto calculates